lib/monitorable_resource.rb

```diff
@@ -41,11 +41,24 @@ def process_event_loop
   def check_pulse
     @run_event_loop = true if @resource.needs_event_loop_for_pulse_check?
 
+    stale_retry = false
     @pulse_check_started_at = Time.now
     begin
       @pulse = @resource.check_pulse(session: @session, previous_pulse: @pulse)
       Clog.emit("Got new pulse.") { {got_pulse: {ubid: @resource.ubid, pulse: @pulse}} } if (rpt = @pulse[:reading_rpt]) && (rpt < 6 || rpt % 5 == 1) || @pulse[:reading] != "up"
     rescue => ex
+      if !stale_retry &&
+          (
+            # Seen when sending on a broken connection.
+            ex.is_a?(IOError) && ex.message == "closed stream" ||
+            # Seen when receiving on a broken connection.
+            ex.is_a?(Errno::ECONNRESET) && ex.message.start_with?("Connection reset by peer - recvfrom(2)")
+          ) &&
+          @session[:last_pulse]&.<(Time.now - 8)
```

Comment:
```suggestion
          @session&.[](:last_pulse)&.<(Time.now - 8)
```

Same issue as in a couple lines down.